### PR TITLE
truncates authorization and cookie values in kitchen request-info response

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- truncates authorization and cookie values in kitchen request-info response

## Why are we making these changes?

We should not display back sensitive information in the response.

Fixes https://github.com/twosigma/waiter/issues/737

### CODE REVIEW

File: containers/test-apps/kitchen/bin/kitchen [hunk 1]
```diff
@@ -121,6 +121,11 @@ def gzip_compress(in_data):
     return out_data
 
 
+def truncate(in_string, truncate_length):
+    "Collapse and truncate the given text to the given length."
+    return in_string[:truncate_length] + (in_string[truncate_length:] and '...')
```
Comment (DaoWen) [inline containers/test-apps/kitchen/bin/kitchen]: I find `(in_string[truncate_length:] and '...')` confusing... Why not `if len(in_string) > truncate_length`? I'd prefer not having the extra slicing operations when that condition fails.

File: containers/test-apps/kitchen/bin/kitchen [hunk 2]
```diff
@@ -943,10 +948,22 @@ class Kitchen(HTTPWebSocketsHandler):
                  'request-length': self.__request_body_length,
                  'request-method': self.__method,
                  'uri': self.__path }
+
         if len(self.headers) > 0:
             info['headers'] = { k.lower(): v for k, v in self.headers.items() }
+
+            cookie_value = info['headers'].get('authorization', None)
```
Comment (DaoWen) [inline containers/test-apps/kitchen/bin/kitchen]: The explicit `None` here is redundant:
https://docs.python.org/3/library/stdtypes.html#dict.get

File: containers/test-apps/kitchen/bin/kitchen [hunk 3]
```diff
@@ -943,10 +948,22 @@ class Kitchen(HTTPWebSocketsHandler):
...
+            if cookie_value is not None:
+                info['headers']['authorization'] = truncate(cookie_value, 15)
+
+            cookie_value = info['headers'].get('cookie', None)
```
Comment (DaoWen) [inline containers/test-apps/kitchen/bin/kitchen]: The explicit `None` here is redundant:
https://docs.python.org/3/library/stdtypes.html#dict.get

File: containers/test-apps/kitchen/bin/kitchen [hunk 4]
```diff
@@ -943,10 +948,22 @@ class Kitchen(HTTPWebSocketsHandler):
...
+            if cookie_value is not None:
+                cookie_entries = [e for e in cookie_value.split(';') if e]
+                info['headers']['cookie'] = [truncate(v, 25) for v in cookie_entries]
```
Comment (DaoWen) [inline containers/test-apps/kitchen/bin/kitchen]: Combine these two lines?
```
info['headers']['cookie'] = [truncate(v, 25) for v in cookie_value.split(';') if v]
```

File: containers/test-apps/kitchen/bin/kitchen (hunk 1 above)
Comment (shamsimam) [reply containers/test-apps/kitchen/bin/kitchen]: Done, uses `if-else` block

File: containers/test-apps/kitchen/bin/kitchen (hunk 2 above)
Comment (shamsimam) [reply containers/test-apps/kitchen/bin/kitchen]: Done, remove `None`.

File: containers/test-apps/kitchen/bin/kitchen (hunk 3 above)
Comment (shamsimam) [reply containers/test-apps/kitchen/bin/kitchen]: Done, remove `None`.

File: containers/test-apps/kitchen/bin/kitchen (hunk 4 above)
Comment (shamsimam) [reply containers/test-apps/kitchen/bin/kitchen]: Combined two lines.

**Pull request merged**